src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -126,6 +110,40 @@ public bool HasValue
         /// <returns>The <see cref="ReadOnlySpan{T}"/> from this <see cref="StringSegment"/>.</returns>
         public ReadOnlySpan<char> AsSpan() => Buffer.AsSpan(Offset, Length);
 
+        /// <summary>
+        /// Gets a <see cref="ReadOnlySpan{T}"/> from the current <see cref="StringSegment"/> that starts
+        /// at the position specified by <paramref name="start"/>, and has the remaining length.
+        /// </summary>
+        /// <param name="start">The zero-based starting character position in this <see cref="StringSegment"/>.</param>
+        /// <returns>A <see cref="ReadOnlySpan{T}"/> with the remaining chars that begins at <paramref name="start"/> in
+        /// this <see cref="StringSegment"/>.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="start"/> is greater than or equal to <see cref="Length"/> or less than zero.
+        /// </exception>
+        public ReadOnlySpan<char> AsSpan(int start) => AsSpan(start, Length - start);
```

Comment:
You can skip the `length` validation here by calling  `Buffer.AsSpan(Offset + start)` directly (and of course doing the HasValue and start validation first).